### PR TITLE
add rp initiated logout redirect

### PR DESCRIPTION
**Pull request by elbotho**
https://serlo.slack.com/archives/C04TTH20JP5/p1682012733496979

### CODE REVIEW

File: src/components/pages/auth/logout.tsx [hunk 1]
```diff
@@ -31,18 +31,25 @@ export function Logout({ oauth }: { oauth?: boolean }) {
   useEffect(() => {
     checkInstance({ redirect: false })
 
-    const redirection = filterUnwantedRedirection({
-      desiredPath: sessionStorage.getItem('previousPathname'),
-      unwantedPaths: [settingsUrl, loginUrl, registrationUrl],
-    })
+    const cookieSession = AuthSessionCookie.parse()
+    const isSSO = cookieSession?.authentication_methods
```
Comment (kulla) [inline src/components/pages/auth/logout.tsx:35]: We can use `io-ts` here (can also be a follow up :smile: )

File: src/components/pages/auth/logout.tsx [hunk 2]
```diff
@@ -31,18 +31,25 @@ export function Logout({ oauth }: { oauth?: boolean }) {
   useEffect(() => {
     checkInstance({ redirect: false })
 
-    const redirection = filterUnwantedRedirection({
-      desiredPath: sessionStorage.getItem('previousPathname'),
-      unwantedPaths: [settingsUrl, loginUrl, registrationUrl],
-    })
+    const cookieSession = AuthSessionCookie.parse()
```
Comment (kulla) [inline src/components/pages/auth/logout.tsx:34]: Is `AuthSessionCookie.parse()` the same as `AuthSessionCookie.get()`?

**Review by kulla — APPROVED** [review]

File: src/components/pages/auth/logout.tsx (hunk 2 above)
Comment (elbotho) [reply src/components/pages/auth/logout.tsx:34]: - `get` returns a `string` 
- `parse` returns a `Session` object (and you can optionally supply a cookie yourself)
But the difference is basically just a `JSON.parse`

File: src/components/pages/auth/logout.tsx (hunk 1 above)
Comment (elbotho) [reply src/components/pages/auth/logout.tsx:35]: Just for line 35-36?
I'd only use it if it's as short tbh :)

**Pull request merged**